test/sdk_test.dart

```diff
@@ -5,222 +5,291 @@
 import 'package:path/path.dart' as p;
 import 'package:pub/src/exit_codes.dart' as exit_codes;
 import 'package:pub/src/io.dart';
+import 'package:pub/src/sdk/sdk_package_config.dart';
 import 'package:test/test.dart';
 
 import 'descriptor.dart' as d;
 import 'test_pub.dart';
 
 void main() {
   forBothPubGetAndUpgrade((command) {
-    setUp(() async {
-      final server = await servePackages();
-      server.serve('bar', '1.0.0');
-
-      await d.dir('flutter', [
-        d.dir('packages', [
-          d.dir('foo', [
-            d.libDir('foo', 'foo 0.0.1'),
-            d.libPubspec('foo', '0.0.1', deps: {'bar': 'any'}),
-          ]),
-        ]),
-        d.dir('bin/cache/pkg', [
-          d.dir(
-            'baz',
-            [d.libDir('baz', 'foo 0.0.1'), d.libPubspec('baz', '0.0.1')],
-          ),
-        ]),
-        d.flutterVersion('1.2.3'),
-      ]).create();
-    });
+    group('flutter', () {
```

Comment:
This diff is kind of insane.... all I actually did was wrap the existing stuff in a group, and make a new group which exercises the Dart sdk support for sdk_packages.yaml.